src/topp/SageAdapter.cpp

```diff
@@ -324,20 +929,51 @@ class TOPPSageAdapter :
 
     config_file.substitute("##enzyme_details##", enzyme_details);
 
+    
     auto fixed_mods = getStringList_("fixed_modifications");
     set<String> fixed_unique(fixed_mods.begin(), fixed_mods.end());
     fixed_mods.assign(fixed_unique.begin(), fixed_unique.end());   
     ModifiedPeptideGenerator::MapToResidueType fixed_mod_map = ModifiedPeptideGenerator::getModifications(fixed_mods); // std::unordered_map<const ResidueModification*, const Residue*> val;
     String static_mods_details = getModDetailsString(fixed_mod_map);
-    config_file.substitute("##static_mods##", static_mods_details);
 
     auto variable_mods = getStringList_("variable_modifications");
     set<String> variable_unique(variable_mods.begin(), variable_mods.end());
     variable_mods.assign(variable_unique.begin(), variable_unique.end());
     ModifiedPeptideGenerator::MapToResidueType variable_mod_map = ModifiedPeptideGenerator::getModifications(variable_mods);
     String variable_mods_details = getModDetailsString(variable_mod_map);
 
-    config_file.substitute("##variable_mods##", variable_mods_details);
+    //Treat variables as list for sage v0.15 and beyond 
```

Comment:
0.14.7?

Reply:
The latest release of Sage is v0.14.7, but the source code from Github is at v0.15.0. From v0.15.0, the old way SageAdapter presented variable modifications to Sage won't work anymore due to the warning being updated to a hard error. I thought I'd add this for future compatibility as well.

Reply:
k fine with me